src/goose.rs

```diff
@@ -1222,6 +1204,29 @@ impl GooseClient {
                     }
                     raw_request.set_status_code(Some(status_code));
                     raw_request.set_final_url(r.url().as_str());
+
+                    // Load test client was redirected.
+                    if raw_request.url != raw_request.final_url {
+                        // Extract the base_url of the request made.
+                        let request_base_url = match Url::parse(&raw_request.url) {
+                            Ok(base_url) => base_url.to_string(),
```

Comment:
Why here the .to_sting() and ...

Reply:
we're using the string to make comparisons (however I'm rewriting this section of the code as it's not properly comparing the base_url)